src/clustering/pqkmeans.cpp

```diff
@@ -141,6 +146,13 @@ std::vector<std::vector<unsigned char>> PQKMeans::GetClusterCenters()
     return centers_;
 }
 
+void PQKMeans::SetClusterCenters(const std::vector<std::vector<unsigned char>> centers_new)
```

Comment:
Put '&' before 'centers_new' to pass the argument as a reference (not copy), i.e., `const std::vector::vector<unsigned char>> &centers_new`